APMrover2/system.cpp

```diff
@@ -386,3 +393,70 @@ bool Rover::disarm_motors(void)
 
     return true;
 }
+
+// position_ok - returns true if the horizontal absolute position is ok and home position is set
+bool Rover::position_ok()
+{
+    // return false if ekf failsafe has triggered // TODO : update with the addition of EKF failsafe
+    /*if (failsafe.ekf) {
+        return false;
+    }*/
+
+    // check ekf position estimate
+    return (ekf_position_ok() || optflow_position_ok());
+}
+
+// ekf_position_ok - returns true if the ekf claims it's horizontal absolute position estimate is ok and home position is set
+bool Rover::ekf_position_ok()
+{
+    if (!ahrs.have_inertial_nav()) {
+        // do not allow navigation with dcm position
+        return false;
+    }
+
+    // with EKF use filter status and ekf check
+    nav_filter_status filt_status = inertial_nav.get_filter_status();
```

Comment:
This requires inertial_nav. And Inertial Nav has been removed in https://github.com/ArduPilot/ardupilot/pull/6720